File: Lyd.MoopDemo/Lyd.MoopDemo.PmsCore/PmsApp.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace Lyd.MoopDemo.PmsCore
{
    public class PmsApp
    {

        private List<Guest> _guests;
        private IGuestCodeGenerator _guestCodeGenerator;

        public PmsApp()
        {
            _guests = new List<Guest>();
            _guestCodeGenerator = new GuestCodeGenerator();
        }


        public List<Guest> InitGuests()
        {
            this._guests.Clear();
            CreateGuest("Chung Vinh Khang", "97 Vo Van Tan", "01227155908", Gender.Male);
            CreateGuest("Bui Cao Tu", "114 Dinh Tien Hoang", "01227155911", Gender.Male);
            CreateGuest("Vu Hoai Nam", "68/8 Ngo Tat To", "01227155308", Gender.Male);
            CreateGuest("Nguyen Duy Phuong", "14 Doan Van Bo", "01227155108", Gender.Male);
            return this._guests;
        }

        public List<Guest> GetGuests()
        {
            return this._guests;
        }

        public List<Guest> SearchGuest(SearchOption searchOption)
        {
            List<Guest> result = _guests.Where(p => p.GuestName.Contains(searchOption.GuestName)).ToList();
            return result;
        }

        public Guest CreateGuest(string guestName, string address, string phone, Gender? gender)
        {
            var newGuest = new Guest(guestName, address, phone, gender);
            newGuest.GenerateGuestCode(_guestCodeGenerator);
            this._guests.Add(newGuest);
            return newGuest;
        }
    }
}

File: Lyd.MoopDemo_Java/pmscore/src/main/java/lyd/demo/pmscore/PmsApp.java
package lyd.demo.pmscore;

import java.util.ArrayList;
import java.util.List;
import java.util.stream.Collectors;

public class PmsApp {
    private List<Guest> guests;
    private GuestCodeGenerator guestCodeGenerator;

    public PmsApp(){
        guests = new ArrayList<>();
        guestCodeGenerator = new GuestCodeGeneratorImpl();
    }

    public List<Guest> initGuests(){
        this.guests.clear();
        this.addGuest("Chung Vinh Khang", "97 Vo Van Tan", "01227xxx5908", Gender.MALE);
        this.addGuest("Tran Dang Khoa", "98 Vo Van Tan", "01227xxx1991", Gender.MALE);
        this.addGuest("Vo Hoang Cham", "99 Vo Van Tan", "0122xxx1999", Gender.FEMALE);
        this.addGuest("Pham Thien An", "96 Vo Van Tan", "0112xxx1918", Gender.FEMALE);
        return this.guests;
    }

    public List<Guest> getGuests(){
        return this.guests;
    }

    public List<Guest> searchGuests(String searchKey){
        return this.guests.stream()
                .filter(p -> p.getGuestName().contains(searchKey))
                .collect(Collectors.toList());
    }

    private Guest addGuest(String guestName, String address, String phone, Gender gender){
        Guest guest = new Guest(guestName, address, phone, gender);
        guest.generateGuestCode(this.guestCodeGenerator);
        this.guests.add(guest);
        return guest;
    }
}
